drone/src/agent/backend.rs

```diff
@@ -107,38 +92,20 @@ impl BackendMonitor {
         })
     }
 
-    fn stats_loop(
+    fn stats_loop<E: Engine>(
         backend_id: &BackendId,
-        docker: &DockerInterface,
+        engine: &E,
         nc: &TypedNats,
     ) -> JoinHandle<Result<(), anyhow::Error>> {
-        let docker = docker.clone();
+        let mut stream = Box::pin(engine.stats_stream(backend_id));
         let nc = nc.clone();
         let backend_id = backend_id.clone();
 
         tokio::spawn(async move {
-            let container_name = backend_id.to_resource_name();
             tracing::info!(%backend_id, "Stats recording loop started.");
-            let mut stream = Box::pin(docker.get_stats(&container_name));
-            let mut prev_stats = stream
-                .next()
-                .await
-                .ok_or_else(|| anyhow!("failed to get first stats"))??;
-            while let Some(cur_stats) = stream.next().await {
-                match cur_stats {
-                    Ok(cur_stats) => {
-                        nc.publish(&BackendStatsMessage::from_stats_messages(
-                            &backend_id,
-                            &prev_stats,
-                            &cur_stats,
-                        )?)
-                        .await?;
-                        prev_stats = cur_stats;
-                    }
-                    Err(error) => {
-                        tracing::warn!(?error, "Error encountered sending stats.")
-                    }
-                }
+
+            while let Some(stats) = stream.next().await {
+                nc.publish(&stats).await?;
```

Comment:
so this error is in a spawn that we don't handle, so does it just quietly drop any error?

Reply:
Changed to log it.